### PR TITLE
zh-trans force-delete-stateful-set-pod.md

### PR DESCRIPTION
**Pull request by mysunshine92**
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Please delete this note before submitting the pull request.
>
> For 1.13 Features: set Milestone to 1.13 and Base Branch to dev-1.13
>
> Help editing and submitting pull requests:
> https://kubernetes.io/docs/contribute/start/#improve-existing-content.
>
> Help choosing which branch to use:
> https://kubernetes.io/docs/contribute/start#choose-which-git-branch-to-use.
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>

### CODE REVIEW

File: content/zh/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+   * Force deletion of the Pod by the user.
+--->
+   * Node 对象被删除(要么您删除, 或者[Node Controller](/docs/admin/node))。<br/>
+   * 无响应节点上的 kubelet 开始响应，杀死 pod 并从 apiserver 中移除该条目。<br/>
+   * 用户强制删除 pod。
+
+<!--
+The recommended best practice is to use the first or second approach. If a Node is confirmed to be dead (e.g. permanently disconnected from the network, powered down, etc), then delete the Node object. If the Node is suffering from a network partition, then try to resolve this or wait for it to resolve. When the partition heals, the kubelet will complete the deletion of the Pod and free up its name in the apiserver.
+--->
+推荐使用第一种或者第二种方法。如果确认节点已经不可用了 (e.g. 永久断开网络，断电等)，则删除 Node 对象。如果节点遇到网络分区，请尝试解决该问题或者等待其解决。当分区愈合时，kubelet 将完成 pod 的删除并从 apiserver 中释放其名字。
```
Comment (tengqm) [inline content/zh/docs/tasks/run-application/force-delete-stateful-set-pod.md]: network partition 不是网络分区，是“网裂”

File: content/zh/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+请始终谨慎地执行强制删除 StatefulSet 类型的 pods，并完全了解所涉及地风险。
+
+{{% /capture %}}
+
+{{% capture whatsnext %}}
+
+<!--
+Learn more about [debugging a StatefulSet](/docs/tasks/debug-application-cluster/debug-stateful-set/).
+--->
+了解更多关于 [调试 StatefulSet](/docs/tasks/debug-application-cluster/debug-stateful-set/)。
```
Comment (tengqm) [inline content/zh/docs/tasks/run-application/force-delete-stateful-set-pod.md]:
```suggestion
进一步了解[调试 StatefulSet](/docs/tasks/debug-application-cluster/debug-stateful-set/)。
```

Comment (mysunshine92) [reply content/zh/docs/tasks/run-application/force-delete-stateful-set-pod.md]: @tengqm 谢谢，都已经修改好了。

File: content/zh/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+<!--
+The recommended best practice is to use the first or second approach. If a Node is confirmed to be dead (e.g. permanently disconnected from the network, powered down, etc), then delete the Node object. If the Node is suffering from a network partition, then try to resolve this or wait for it to resolve. When the partition heals, the kubelet will complete the deletion of the Pod and free up its name in the apiserver.
+--->
+推荐使用第一种或者第二种方法。如果确认节点已经不可用了 (比如，永久断开网络，断电等)，则删除 Node 对象。如果节点遇到网裂，请尝试解决该问题或者等待其解决。当网裂愈合时，kubelet 将完成 pod 的删除并从 apiserver 中释放其名字。
+
+<!--
+Normally, the system completes the deletion once the Pod is no longer running on a Node, or the Node is deleted by an administrator. You may override this by force deleting the Pod.
+--->
+通常，pod 一旦不在节点上运行，或者管理员删除了节点，系统就会完成删除。你可以通过强制删除 pod 来覆盖它。
+### 强制删除
```
Comment (markthink) [inline content/zh/docs/tasks/run-application/force-delete-stateful-set-pod.md]: 这里原始的英文呢？

Comment (mysunshine92) [reply content/zh/docs/tasks/run-application/force-delete-stateful-set-pod.md]: @markthink 谢谢，已经修改好了，所有地方都保留了原始的英文。

File: content/zh/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+您可以使用下面的命令执行优雅地删除 pod:
+
+```shell
+kubectl delete pods <pod>
+```
+
+<!--
+For the above to lead to graceful termination, the Pod **must not** specify a `pod.Spec.TerminationGracePeriodSeconds` of 0. The practice of setting a `pod.Spec.TerminationGracePeriodSeconds` of 0 seconds is unsafe and strongly discouraged for StatefulSet Pods. Graceful deletion is safe and will ensure that the [Pod shuts down gracefully](/docs/user-guide/pods/#termination-of-pods) before the kubelet deletes the name from the apiserver.
+--->
+为了使上面的方法能够正常终止，Pod **一定不能**设置  `pod.Spec.TerminationGracePeriodSeconds` 为0。将 `pod.Spec.TerminationGracePeriodSeconds` 设置为 0s 的做法是不安全的，强烈建议 StatefulSet 类型的 pods 不要使用。优雅删除是安全的，并且会在 kubelet 从 apiserver 中删除名称之前确保 [ pod 是优雅关闭的](/docs/user-guide/pods/#termination-of-pods)。
```
Comment (pigletfly) [inline content/zh/docs/tasks/run-application/force-delete-stateful-set-pod.md]:
```suggestion
为了使上面的方法能够正常终止，Pod **一定不能**设置  `pod.Spec.TerminationGracePeriodSeconds` 为 0。将 `pod.Spec.TerminationGracePeriodSeconds` 设置为 0s 的做法是不安全的，强烈建议 StatefulSet 类型的 pods 不要使用。优雅删除是安全的，并且会在 kubelet 从 apiserver 中删除名称之前确保 [优雅地关闭 pod ](/docs/user-guide/pods/#termination-of-pods)。
```

Comment (mysunshine92) [reply content/zh/docs/tasks/run-application/force-delete-stateful-set-pod.md]: @pigletfly 谢谢您的评审意见，已经修改好了。

**Comment by mysunshine92** [comment]
@markthink would you like to add lgtm label? thank you

**Pull request merged**